### PR TITLE
fix long bar version

### PR DESCRIPTION
**Pull request by darasus**
- fix long bar version ([issue](https://github.com/pastelsky/bundlephobia/issues/684))

CleanShot 2022-11-03 at 08 21 35@2x

### CODE REVIEW

File: client/components/BarVersion/BarVersion.scss [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    display: flex;
+    text-align: end;
+    justify-content: flex-start;
+    align-items: center;
+    height: 100%;
+    width: 100%;
+    > span {
+      white-space: nowrap;
+      overflow: hidden;
+      text-overflow: ellipsis;
```
Comment (pastelsky) [inline client/components/BarVersion/BarVersion.scss:35]: The ellipsis part doesn't seem to work in Safari.

File: client/components/BarVersion/BarVersion.scss [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  > div {
+    display: flex;
+    text-align: end;
+    justify-content: flex-start;
+    align-items: center;
+    height: 100%;
+    width: 100%;
+    > span {
+      white-space: nowrap;
+      overflow: hidden;
```
Comment (pastelsky) [inline client/components/BarVersion/BarVersion.scss:34]: Let's add a title attribute so folks can hover over to see the full version if they wish to.

File: client/components/BarVersion/BarVersion.scss (hunk 1 above)
Comment (darasus) [reply client/components/BarVersion/BarVersion.scss:35]: CleanShot 2022-11-03 at 16 01 52@2x

Looks okay to me?

Comment (darasus) [reply client/components/BarVersion/BarVersion.scss:35]: Also here seems to have support https://caniuse.com/?search=ellipsis

File: client/components/BarVersion/BarVersion.scss [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  color: lighten($raven, 15%);
+  transition: opacity 0.2s, color 0.2s;
+  font-family: $font-family-code;
+  letter-spacing: -1px;
+  line-height: 1;
+  cursor: pointer;
+  height: $bar-width;
+  writing-mode: vertical-rl;
+  text-orientation: sideways;
+  height: 50px;
```
Comment (pastelsky) [inline client/components/BarVersion/BarVersion.scss:19]: There's already a height attribute above (on that note, we should probably add stylelint as well to this project).

File: client/components/BarVersion/BarVersion.scss [hunk 4]
```diff
@@ -0,0 +1,46 @@
+@import '../../../stylesheets/colors';
+@import '../../../stylesheets/variables';
+
+.bar-graph__bar-version {
```
Comment (pastelsky) [inline client/components/BarVersion/BarVersion.scss:4]: This solution produces a weird alignment for a package like `react-pdf@4.2.0-beta`

Screenshot 2022-11-03 at 8 32 04 PM

Ideally all of these versions would stick up to the top? Arguable.

File: client/components/BarVersion/BarVersion.scss (hunk 1 above)
Comment (pastelsky) [reply client/components/BarVersion/BarVersion.scss:35]: You need to test with a package with long versions like - `@iconify-prerendered/vue-mdi` to repro.
 If its too hard to get it to work for safari, I wouldn't spend too much time on it.

File: client/components/BarVersion/BarVersion.scss [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+  line-height: 1;
+  cursor: pointer;
+  height: $bar-width;
+  writing-mode: vertical-rl;
+  text-orientation: sideways;
+  height: 50px;
+  text-align: left;
+  justify-content: flex-start;
+  display: inline-block;
+  margin-top: 5px;
```
Comment (pastelsky) [inline client/components/BarVersion/BarVersion.scss:23]: These properties seem to be a no-op —

```css
text-orientation: sideways;
text-align: left; 
justify-content: flex-start; 
display: inline-block;
```

File: client/components/BarVersion/BarVersion.scss [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+
+  > div {
+    display: flex;
+    text-align: end;
+    justify-content: flex-start;
+    align-items: center;
+    height: 100%;
+    width: 100%;
```
Comment (pastelsky) [inline client/components/BarVersion/BarVersion.scss:31]: These properties seem to be a no-op —

```css
text-align: end;
justify-content: flex-start;
height: 100%;
```

**Review by pastelsky — CHANGES_REQUESTED** [review]

**Comment by darasus** [comment]
@pastelsky we can also do truncation with javascript, check last commit

**Review by pastelsky — APPROVED** [review]

**Pull request merged**